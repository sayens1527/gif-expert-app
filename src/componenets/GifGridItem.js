import React from 'react'

const GifGridItem = ({id, title, url}) => {

    return (
        <div className='grid-item-item'>
            <p> { title } </p>
            <img src={url} alt= {title}></img>
        </div>
    )
}

GifGridItem.propTypes = {

}

export default GifGridItem
