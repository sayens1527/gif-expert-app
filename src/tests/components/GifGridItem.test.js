import {shallow} from 'enzyme';
import React from 'react';
import GifGridItem from '../../components/GifGridItem'


describe('Test in GifGridItem component', () => {

    
    /*beforeEach( () => {
        wrapper = shallow(<CounterApp/>);
    });*/

    const url = 'url';
    const title = 'title';
    const wraper = shallow(<GifGridItem title={ title} url={ url }></GifGridItem>);
    
    test('should match with the snapshot', () => {
        expect(wraper).toMatchSnapshot();
    });

    test('should show the title in p element', () => {

        const titleInComponent = wraper.find('p').text().trim();
        expect(titleInComponent).toBe(title);

    });
    
    test('should show the image whit its attributes', () => {

        const img = wraper.find('img');
        const imgSrc =  img.prop('src');
        const imgAlt = img.prop('alt');

        expect (imgSrc).toBe(url);
        expect (imgAlt).toBe(title);
        
    });

    test('should has the class grid-item-item', () => {

        const div = wraper.find('div');

        const divClass = div.prop('className');

        expect(divClass.includes('grid-item-item') ).toBe(true);
        
    });
    
    
    
})
