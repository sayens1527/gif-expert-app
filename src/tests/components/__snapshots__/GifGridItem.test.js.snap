// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifGridItem component should match with the snapshot 1`] = `
<div
  className="grid-item-item"
>
  <p>
     
    title
     
  </p>
  <img
    alt="title"
    src="url"
  />
</div>
`;
